myapp/backend/backend.py

```diff
@@ -220,6 +202,30 @@ def get_team_roster():
     return resp
 
 
+def roster_get_link_parse(name, status, position, specialization):
+    filters = {}
+
+    single_name = None
+    if name is not None:
+        split_names = name.split()
+
+        if len(split_names) == 2:
+            filters['member_first_name'] = {'$regex': f'{split_names[0]}', '$options': 'i'}
+            filters['member_last_name'] = {'$regex': f'{split_names[1]}', '$options': 'i'}
+            single_name = None
+        else:
+            single_name = split_names[0]
+
+    if status is not None:
+        filters['member_status'] = status
+    if position is not None:
+        filters['member_position'] = position
+    if specialization is not None:
+        filters['member_specialization'] = {'$in': [f'{specialization}']}
+
+    return filters, single_name
+
```

Comment:
I like how you split it up into its own function! I really like that and now it's more readable!

Reply:
Agreed!